Automattic/jetpack
# Admin Page: add custom numberFormat implementation

#### Changes proposed in this Pull Request:

This would allow us to get rid of the i18n-calypso package; the only reason we keep this package around today is for this:
https://github.com/Automattic/jetpack/pull/16397#issuecomment-702135251

This implementation relies on the browser to do all the work, based on the locale given by the browser or by WordPress.

<img width="617" alt="screenshot 2020-10-02 at 18 10 48" src="https://user-images.githubusercontent.com/426388/94945439-f1bcdc80-04da-11eb-831d-f4a82f81a5b0.png">
<img width="604" alt="screenshot 2020-10-02 at 17 43 21" src="https://user-images.githubusercontent.com/426388/94945442-f4b7cd00-04da-11eb-9a27-f57c02635bf2.png">
<img width="615" alt="screenshot 2020-10-02 at 17 42 43" src="https://user-images.githubusercontent.com/426388/94945448-f5506380-04da-11eb-9d57-4f76c5823357.png">

#### Jetpack product discussion

Related issues: #16481, #16397

#### Does this pull request change what data or activity we track or use?

* No 

#### Testing instructions:

* Go to Jetpack > Dashboard
* Look at At a Glance numbers; they should all be displayed according to the standard expected in your locale.
* Try disconnecting your site from WordPress.com to see even more numbers; they should all be displayed properly.
* Try switching to a different site language under Settings > General. You should see different results based on the language. (I would encourage you to test in locale variants as well, such as "Deutsch (Schweiz, Du)").

#### Proposed changelog entry for your changes:

* N/A
